java/memory/src/main/java/org/apache/arrow/memory/NettyAllocationManager.java

```diff
@@ -17,48 +17,97 @@
+   * {@link PooledByteBufAllocatorL} APIs,
+   * otherwise, we will use {@link PlatformDependent} APIs.
+   */
+  public static final long DEFAULT_ALLOCATION_CUTOFF_VALUE;
+
+  public static final String DEFAULT_ALLOCATION_CUTOFF_NAME = "default.allocation.cutoff.name";
+
+  static {
+    long cutOffValue;
+    try {
+      cutOffValue = Long.parseLong(System.getProperty(DEFAULT_ALLOCATION_CUTOFF_NAME));
+    } catch (Exception e) {
+      cutOffValue = Integer.MAX_VALUE;
+    }
+    DEFAULT_ALLOCATION_CUTOFF_VALUE = cutOffValue;
+  }
+
   private static final PooledByteBufAllocatorL INNER_ALLOCATOR = new PooledByteBufAllocatorL();
   static final UnsafeDirectLittleEndian EMPTY = INNER_ALLOCATOR.empty;
   static final long CHUNK_SIZE = INNER_ALLOCATOR.getChunkSize();
 
-  private final int allocatedSize;
+  private final long allocatedSize;
   private final UnsafeDirectLittleEndian memoryChunk;
+  private final long allocatedAddress;
+
+  /**
+   * The cut-off value for switching allocation strategies.
+   */
+  private final long allocationCutOffValue;
 
-  NettyAllocationManager(BaseAllocator accountingAllocator, int requestedSize) {
+  NettyAllocationManager(BaseAllocator accountingAllocator, long requestedSize, long allocationCutOffValue) {
     super(accountingAllocator);
-    this.memoryChunk = INNER_ALLOCATOR.allocate(requestedSize);
-    this.allocatedSize = memoryChunk.capacity();
+    if (allocationCutOffValue > Integer.MAX_VALUE) {
+      throw new IllegalArgumentException("The cut-off value cannot be larger than Integer.MAX_VALUE");
+    }
+    this.allocationCutOffValue = allocationCutOffValue;
+
+    if (requestedSize > allocationCutOffValue) {
+      this.memoryChunk = null;
+      this.allocatedAddress = PlatformDependent.allocateMemory(requestedSize);
+      this.allocatedSize = requestedSize;
+    } else {
+      this.memoryChunk = INNER_ALLOCATOR.allocate(requestedSize);
+      this.allocatedAddress = memoryChunk.memoryAddress();
+      this.allocatedSize = memoryChunk.capacity();
+    }
+  }
+
+  NettyAllocationManager(BaseAllocator accountingAllocator, long requestedSize) {
+    this(accountingAllocator, requestedSize, DEFAULT_ALLOCATION_CUTOFF_VALUE);
   }
 
   /**
    * Get the underlying memory chunk managed by this AllocationManager.
-   * @return buffer
+   * @return the underlying memory chunk if the request size is not greater than the
+   *   {@link NettyAllocationManager#allocationCutOffValue}, or null otherwise.
+   *
+   * @deprecated this method will be removed in a future release.
    */
+  @Deprecated
   UnsafeDirectLittleEndian getMemoryChunk() {
-    return memoryChunk;
+    return allocatedSize > allocationCutOffValue ? null : memoryChunk;
   }
 
   @Override
   protected long memoryAddress() {
-    return memoryChunk.memoryAddress();
+    return allocatedAddress;
   }
 
   @Override
   protected void release0() {
-    memoryChunk.release();
+    if (allocatedSize > allocationCutOffValue) {
```

Comment:
just check if `memoryChunk` is null here

Reply:
Revised accordingly. Thank you.